### PR TITLE
Support configuring the UI page title

### PR DESCRIPTION
**Pull request by gabrielmaldi**
**What this PR does / why we need it**:

Adds an option to configure the UI page title in `<head>`.

**Which issue(s) this PR fixes**:
> Change the title of the web interface

Please reference the issue this PR will close: #501

**Special notes for your reviewer**:

I didn't add documentation because currently similar options like `AsideMenuOpened` and others aren't documented, so I assumed this is not an important enough option, and it's discoverable the same way `AsideMenuOpened` is.

Nevertheless, a usage example is included in the sample project.

**Does this PR introduce a user-facing change?**:

Yes

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [x] Created/updated tests
- [x] Unit tests passing
- [x] End-to-end tests passing
- [ ] Extended the documentation
- [x] Provided sample for the feature

---

This PR is based on #696 with a test added. Credits to @MCKanpolat!

### CODE REVIEW

**Comment by unaizorrilla** [comment]
Thanks for summit this PR!

@CarlosLanderas  can you take a look on this?

**Comment by unaizorrilla** [comment]
Hi  @CarlosLanderas , can you take a look

**Comment by dotnetmat90** [comment]
Any news when we can expect it?

**Comment by sungam3r** [comment]
@gabrielmaldi Please rebase onto `master` so CI can verify your changes.

**Comment by sungam3r** [comment]
Then I can review/merge. I see changes are simple so it will take a little to review.

**Comment by gabrielmaldi** [comment]
@sungam3r, done 💪🏻

File: samples/HealthChecks.UIAndApiCustomization/Startup.cs [hunk 1]
```diff
@@ -64,6 +64,7 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
                     {
                         setup.UIPath = "/show-health-ui"; // this is ui path in your browser
                         setup.ApiPath = "/health-ui-api"; // the UI ( spa app )  use this path to get information from the store ( this is NOT the healthz path, is internal ui api )
+                        setup.PageTitle = "Health Checks UI"; // the page title in <head>
```
Comment (sungam3r) [inline samples/HealthChecks.UIAndApiCustomization/Startup.cs:67]:
```suggestion
                        setup.PageTitle = "My wonderful Health Checks UI"; // the page title in <head>

```

Just to show the title differ from the standard one. OK?

File: test/FunctionalTests/HealthChecks.UI/Configuration/UIConfigurationTests.cs [hunk 2]
```diff
@@ -319,5 +321,39 @@ public void allow_disable_running_database_migrations_using_configuration_provid
...
             UISettings.DisableMigrations.Should().Be(true);
         }
+
+        [Fact]
+        public async Task support_configuring_page_title()
+        {
+            var pageTitle = "My Health Checks UI";
+
+            var builder = new WebHostBuilder()
+                .UseStartup<DefaultStartup>()
```
Comment (sungam3r) [inline test/FunctionalTests/HealthChecks.UI/Configuration/UIConfigurationTests.cs:331]: 😕  A "small" problem here. I removed `DefaultStartup` class some time ago so this code does not compile...
Actually `FunctionalTests.csproj` was lost from sight. Maybe `HealthChecks.UI.Tests` is a right place to go (most likely). I don't know exactly but it seems that project was renamed/copied some time ago. Also tests should be enabled for UI workflow here https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/master/.github/workflows/healthchecks_ui_ci.yml#L148 . Now the tests do not start at all.

ping @carlosrecuero 

@gabrielmaldi Small changes lead to great work 😉 .

Comment (gabrielmaldi) [reply test/FunctionalTests/HealthChecks.UI/Configuration/UIConfigurationTests.cs]: Done.

File: samples/HealthChecks.UIAndApiCustomization/Startup.cs (hunk 1 above)
Comment (gabrielmaldi) [reply samples/HealthChecks.UIAndApiCustomization/Startup.cs]: Done.

File: test/FunctionalTests/HealthChecks.UI/Configuration/UIConfigurationTests.cs (hunk 2 above)
Comment (gabrielmaldi) [reply test/FunctionalTests/HealthChecks.UI/Configuration/UIConfigurationTests.cs]: I hadn't noticed all this, I just rebased and pushed, sorry!

Now I moved my new test to the right project, deleted the obsolete FunctionalTests folder (in a separate commit if you wish to revert it), and enabled the UI workflow tests in the file you pointed to.

Thanks!

Comment (sungam3r) [reply test/FunctionalTests/HealthChecks.UI/Configuration/UIConfigurationTests.cs]: UI tests hang. I have not yet found an answer.

**Comment by gabrielmaldi** [comment]
Now we now why the UI tests were disabled 😅. I don't think this PR is the right place to fix all of those tests. Maybe we should just disable them again?

**Comment by sungam3r** [comment]
I'm working on tests in #997

**Comment by sungam3r** [comment]
I think you may revert test changes and I'' merge as is.

**Comment by sungam3r** [comment]
By the way, `support_configuring_page_title` test should fail with error regarding no startup class has been registered.

**Comment by gabrielmaldi** [comment]
> I think you may revert test changes and I'll merge as is.

Should I also revert the `Remove obsolete FunctionalTests` commit or just `Enable tests for UI workflow`?

> By the way, `support_configuring_page_title` test should fail with error regarding no startup class has been registered.

```
Error Message:
   System.InvalidOperationException : No application configured. Please specify startup via IWebHostBuilder.UseStartup, IWebHostBuilder.Configure, injecting IStartup or specifying the startup assembly via StartupAssemblyKey in the web host configuration.
  Stack Trace:
     at Microsoft.AspNetCore.Hosting.WebHost.EnsureStartup()
   at Microsoft.AspNetCore.Hosting.WebHost.EnsureApplicationServices()
   at Microsoft.AspNetCore.Hosting.WebHost.Initialize()
   at Microsoft.AspNetCore.Hosting.WebHostBuilder.Build()
   at HealthChecks.UI.Tests.UI_Configuration_should.should_register_configured_http_client_and_http_message_handler_into_client() in /_/test/HealthChecks.UI.Tests/Functional/Configuration/UIConfigurationTests.cs:line 153
  Failed HealthChecks.UI.Tests.UI_Configuration_should.support_configuring_page_title [5 ms]
```

https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/runs/5516334613?check_suite_focus=true#step:32:635

**Comment by sungam3r** [comment]
I suggest revert removal of "old" tests and enabling "new" UI tests.

**Comment by gabrielmaldi** [comment]
Done!

**Comment by sungam3r** [comment]
OK, I'll merge this when it is completed.

**Comment by gabrielmaldi** [comment]
Great, @sungam3r, thank you for your time!

**Comment by sungam3r** [comment]
I do not think that this feature will be released before fix for UI tests.

**Pull request merged**